pkg/filemap/filemap.go

```diff
@@ -165,19 +171,49 @@ func (fm *Filemap) EncodeToInputText() (string, error) {
 }
 
 // EncodeToInputTextFullPaths Encodes the filemap into a string using each file's full path as its tagname.
-func (fm *Filemap) EncodeToInputTextFullPaths() (string, error) {
+func (fm *Filemap) EncodeToInputTextFullPaths(outputType string) (string, error) {
 	var input string = ""
 	var i int = 0
+	var GenFiles []openai.FileOutput
+
 	// join the files together along with their tag
 	for _, file := range fm.Files {
-		input += fmt.Sprintf("# %s%s\n%s\n", FILE_TAG_PREFIX, file.Path, file.Content)
+
+		sinFile := openai.FileOutput{
+			Name:     file.Name,
+			Path:     file.Path,
+			Contents: file.Content,
+		}
+
+		GenFiles = append(GenFiles, sinFile)
+
+		input += fmt.Sprintf("# %s%s\n%s\n", FILE_TAG_PREFIX, sinFile.Path, sinFile.Contents)
 		// insert a delimeter between each file, but not after the last file
 		if 1 < len(fm.Files) && i < len(fm.Files)-1 {
 			input += fmt.Sprintf("%s\n", FILE_DELIMETER)
 		}
 		i++
 	}
-	return input, nil
+
+	GeneratedFiles := openai.GeneratedFilesOutput{
+		GeneratedFiles: GenFiles,
+	}
+
+	genFilesOut, err := json.MarshalIndent(GeneratedFiles, "", "    ")
+	if err != nil {
+		return "", err
+	}
+
+	if outputType == "json" {
```

Comment:
you'll probably want to turn these into const values and place them at the top of the file like this:
```go
const (
	OutputJSON  = "json"
	OutputPlain = "plain"
)
```
This way, I can easily reference it anywhere that we use the filemap:
```go

// somewhere else in the code
output, err := EncodeToInputTextFullPaths(filemap.OutputJSON)
```